internal/backend/runtime/omni/controllers/omni/versions.go

```diff
@@ -113,20 +114,26 @@ var allowedPreVersionStrings = map[string]struct{}{
 	"beta":  {},
 }
 
-func (ctrl *VersionsController) getVersionsAfter(ctx context.Context, source string, minVersion semver.Version, includePreReleaseVersions bool) ([]string, error) {
-	versions, err := registry.UpgradeCandidates(ctx, source)
+func (ctrl *VersionsController) fetchVersionsFromRegistry(ctx context.Context, source string) ([]string, error) {
+	return registry.UpgradeCandidates(ctx, source)
+}
+
+func (ctrl *VersionsController) fetchTalosVersions(ctx context.Context, factoryURL string) ([]string, error) {
+	client, err := client.New(factoryURL)
 	if err != nil {
 		return nil, err
 	}
 
+	return client.Versions(ctx)
```

Comment:
do we also need to fetch this periodically or that happens already?

Reply:
that happens already, we just change the source here